Calendars/Calendars.Plugin.Tizen/CalendarsImplementation.cs

```diff
@@ -0,0 +1,437 @@
+			Crecord.Set<string>(Book.Name, "org.tizen.calendar");
+
+			Calendar calendar = new Calendar
+			{
+				Name = Crecord.Get<string>(Book.Name),
+				ExternalID = Crecord.Get<int>(Book.Id).ToString(),
+				CanEditEvents = true,
+				CanEditCalendar = false,
+				AccountName = Crecord.Get<int>(Book.AccountId).ToString(),
+				Color = Crecord.Get<string>(Book.Color),
+			};
+
+			Crecord?.Dispose();
+			Crecord = null;
+
+			return Task.Run(() => calendar);
+		}
+
+		/// <summary>
+		/// Gets all events for a calendar within the specified time range.
+		/// </summary>
+		/// <param name="calendar">Calendar containing events</param>
+		/// <param name="start">Start of event range</param>
+		/// <param name="end">End of event range</param>
+		/// <returns>Calendar events</returns>
+		/// <exception cref="System.ArgumentException">Calendar does not exist on device</exception>
+		/// <exception cref="System.UnauthorizedAccessException">Calendar access denied</exception>
+		/// <exception cref="Plugin.Calendars.Abstractions.PlatformException">Unexpected platform-specific error</exception>
+		public Task<IList<CalendarEvent>> GetEventsAsync(Calendar calendar, DateTime start, DateTime end)
+		{
+			return Task.Run<IList<CalendarEvent>>(() =>
+			{
+				var calendars = new List<CalendarEvent>();
+
+				CalendarManager Cmanager = new CalendarManager();
+				CalendarList Clist = Cmanager.Database.GetAll(Event.Uri, 0, 0);
+				CalendarRecord Crecord = null;
+
+				if (Clist.Count == 0)
+					return calendars;
+
+				Clist.MoveFirst();
+
+				TimeSpan utc = new TimeSpan(9, 0, 0);
+				do
+				{
+					Crecord = Clist.GetCurrentRecord();
+
+					string Summary = Crecord.Get<string>(Event.Summary);
+					int IsAllday = Crecord.Get<int>(Event.IsAllday);
+					CalendarTime Start = Convert.ToBoolean(IsAllday)
+							? ConvertIntPtrToCalendarTime(new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc))
+							: Crecord.Get<CalendarTime>(Event.Start);
+					CalendarTime End = Convert.ToBoolean(IsAllday)
+									? ConvertIntPtrToCalendarTime(new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc))
+									: Crecord.Get<CalendarTime>(Event.End);
+					string Location = Crecord.Get<string>(Event.Location);
+					string Description = Crecord.Get<string>(Event.Description);
+					int Id = Crecord.Get<int>(Event.Id);
+
+					TimeSpan s = Start.UtcTime - start + utc;
+					TimeSpan e = end - End.UtcTime + utc;
+					if (s.TotalSeconds >= 0 && e.TotalSeconds >= 0)
+					{
+						calendars.Add(
+							new CalendarEvent
+							{
+								Name = Summary,
+								Start = Start.UtcTime + utc,
+								End = End.UtcTime + utc,
+								Location = Location,
+								Description = Description,
+								AllDay = Convert.ToBoolean(IsAllday),
+								ExternalID = Id.ToString(),
+							}
+						);
+					}
+				} while (Clist.MoveNext());
+
+				Crecord?.Dispose();
+				Crecord = null;
+
+				Clist?.Dispose();
+				Clist = null;
+
+				Cmanager?.Dispose();
+				Cmanager = null;
```

Comment:
You may want to consider `using` blocks instead of manually calling `Dispose` on everything at the end... especially if there's a chance of exceptions being thrown and things thus not getting disposed...